### PR TITLE
#781: Audit `QWATCH` documentation for consistency

### PR DESCRIPTION
**Pull request by shashank-priyadarshi**
Fixes [QWATCH audit and doc update](https://github.com/DiceDB/dice/issues/781).

- Refactor using [SET Command docs](https://dicedb.io/commands/set/) as reference
- Use $ for path demonstration instead of .

### CODE REVIEW

File: docs/src/content/docs/commands/QWATCH.md [hunk 1]
```diff
@@ -3,81 +3,123 @@ title: QWATCH
...
+| Parameter    | Description                                                                         | Type   | Required |
+|--------------|-------------------------------------------------------------------------------------|--------|----------|
+| `dsql-query` | A SQL-like query specifying the data to be monitored and operation to be performed. | String | Yes      |
 
-  - `SELECT`: Specifies the fields to be returned, `$key`, `$value`, and `$value.<attr>`.
-  - `WHERE`: Optional clause for filtering results based on conditions.
-  - `LIKE`: Optional clause within WHERE to specify the key pattern and supports the `%` operator from SQL.
-  - `ORDER BY`: Optional clause for sorting results.
-  - `LIMIT`: Optional clause to limit the number of results.
+## DSQL Query
```
Comment (JyotinderSingh) [inline docs/src/content/docs/commands/QWATCH.md:35]: No need to remove this.

File: docs/src/content/docs/commands/QWATCH.md [hunk 2]
```diff
@@ -87,70 +129,81 @@ This query does the following:
...
-## Error Handling
+   QWATCH Response:
+   ```bash 
+   qwatch    from SELECT $key, $value WHERE $key like 'match:100:*' and $value > 100 ORDER BY $value asc: `[["match:100:user:0", "30"], ["match:100:user:4", "25"], ["match:100:user:2", "20"]]`
+   ```
 
-- `ERR invalid query`: If the provided query is malformed or unsupported.
-- `ERR syntax error`: If the query syntax is incorrect.
-- `ERR unknown command`: If the `QWATCH` command is not implemented.
-- `ERR max number of subscriptions reached`: If the maximum number of allowed subscriptions is exceeded.
```
Comment (JyotinderSingh) [inline docs/src/content/docs/commands/QWATCH.md:153]: We shouldn't remove this. Could you please help verify what different kinds of major errors can be returned and add them here?

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]
Thanks for these change, have left a few comments.

File: docs/src/content/docs/commands/QWATCH.md (hunk 1 above)
Comment (shashank-priyadarshi) [reply docs/src/content/docs/commands/QWATCH.md:35]: Refactored this to this: https://github.com/shashank-priyadarshi/dice/blob/851a9d4cbc57a878bfe484e6f213c4de12e7b872/docs/src/content/docs/commands/QWATCH.md?plain=1#L34

File: docs/src/content/docs/commands/QWATCH.md (hunk 2 above)
Comment (shashank-priyadarshi) [reply docs/src/content/docs/commands/QWATCH.md:153]: Haven't removed it either. Please check out this link: https://github.com/shashank-priyadarshi/dice/blob/851a9d4cbc57a878bfe484e6f213c4de12e7b872/docs/src/content/docs/commands/QWATCH.md?plain=1#L96

**Comment by shashank-priyadarshi** [comment]
@JyotinderSingh went through the review comments. Can you please review and let me know if I still need to update the pull request!

File: docs/src/content/docs/commands/QWATCH.md [hunk 3]
```diff
@@ -3,81 +3,123 @@ title: QWATCH
...
+### Syntax
 
 ```sql
-SELECT $key, $value
-WHERE condition
-ORDER BY field [ASC | DESC] LIMIT n
+SELECT $key,
+       $value WHERE condition [AND | OR] $key LIKE 'pattern' [AND | OR] ($key != n [AND | OR] $value > n)
+ORDER BY field [ASC | DESC]
+    LIMIT n
```
Comment (JyotinderSingh) [inline docs/src/content/docs/commands/QWATCH.md:44]: This isn't the correct grammar for the command. Also, we haven't yet defined a formal grammar, let's stick with the existing syntax:
```sql
SELECT $key, $value
WHERE condition
ORDER BY field [ASC | DESC] LIMIT n
```

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]

File: docs/src/content/docs/commands/QWATCH.md (hunk 3 above)
Comment (shashank-priyadarshi) [reply docs/src/content/docs/commands/QWATCH.md]: Reverted!

**Comment by shashank-priyadarshi** [comment]
@JyotinderSingh reverted the change to syntax example. Please let me know if further changes are required.

**Review by JyotinderSingh — APPROVED** [review]
Thanks for these changes, approved.

**Pull request merged**